C++20 support

**Description**

```
../apps/websocket_node/foxglove_websocket/include/foxglove/websocket/websocket_server.hpp:266:37: error: aggregate initialization of type 'foxglove::Server<foxglove::WebSocketNoTls>::ClientInfo' with user-declared constructors is incompatible with C++20 [-Werror,-Wc++20-compat]
    _clients.emplace(hdl, ClientInfo{endpoint, hdl, {}, {}});
                                    ^~~~~~~~~~~~~~~~~~~~~~~
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -81,3 +81,4 @@ jobs:
       - uses: actions/checkout@v3
       - run: make format-check
       - run: make build
+      - run: make build-cpp20

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -81,4 +81,12 @@ jobs:
       - uses: actions/checkout@v3
       - run: make format-check
       - run: make build
+
+  cpp-std-20:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: cpp
+    steps:
+      - uses: actions/checkout@v3
       - run: make build-cpp20

--- a/cpp/Makefile
+++ b/cpp/Makefile
@@ -2,7 +2,11 @@ default: build
 
 .PHONY: build
 build:
-	docker compose build
+	docker compose build --build-arg CPPSTD=17
+
+.PHONY: build-cpp20
+build:
+	docker compose build --build-arg CPPSTD=20
 
 .PHONY: format-check
 format-check:

--- a/cpp/Makefile
+++ b/cpp/Makefile
@@ -5,7 +5,7 @@ build:
 	docker compose build --build-arg CPPSTD=17
 
 .PHONY: build-cpp20
-build:
+build-cpp20:
 	docker compose build --build-arg CPPSTD=20
 
 .PHONY: format-check

--- a/cpp/dev.Dockerfile
+++ b/cpp/dev.Dockerfile
@@ -27,15 +27,16 @@ RUN conan profile detect --force
 
 FROM build as build_example_server
 COPY ./foxglove-websocket/conanfile.py /src/foxglove-websocket/conanfile.py
-RUN conan install foxglove-websocket -s compiler.cppstd=17 --build=missing
+ARG CPPSTD=17
+RUN conan install foxglove-websocket -s compiler.cppstd=$CPPSTD --build=missing
 COPY ./foxglove-websocket /src/foxglove-websocket/
-RUN conan create foxglove-websocket -s compiler.cppstd=17
+RUN conan create foxglove-websocket -s compiler.cppstd=$CPPSTD
 COPY ./examples/conanfile.py /src/examples/conanfile.py
-RUN conan install examples --output-folder examples/build --build=missing -s compiler.cppstd=17
+RUN conan install examples --output-folder examples/build --build=missing -s compiler.cppstd=$CPPSTD
 
 FROM build_example_server AS example_server
 COPY --from=build_example_server /src /src
 COPY ./examples /src/examples
 COPY --from=build_example_server /src/examples/build/ /src/examples/build/
-RUN conan build examples --output-folder examples/ -s compiler.cppstd=17
+RUN conan build examples --output-folder examples/ -s compiler.cppstd=$CPPSTD
 CMD ["examples/build/Release/example_server"]

--- a/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp
+++ b/cpp/foxglove-websocket/include/foxglove/websocket/websocket_server.hpp
@@ -148,6 +148,10 @@ private:
     std::unordered_set<ClientChannelId> advertisedChannels;
     bool subscribedToConnectionGraph = false;
 
+    explicit ClientInfo(const std::string& name, ConnHandle handle)
+        : name(name)
+        , handle(handle) {}
+
     ClientInfo(const ClientInfo&) = delete;
     ClientInfo& operator=(const ClientInfo&) = delete;
 
@@ -270,7 +274,7 @@ inline void Server<ServerConfiguration>::handleConnectionOpened(ConnHandle hdl) 
 
   {
     std::unique_lock<std::shared_mutex> lock(_clientsMutex);
-    _clients.emplace(hdl, ClientInfo{endpoint, hdl, {}, {}});
+    _clients.emplace(hdl, ClientInfo(endpoint, hdl));
   }
 
   con->send(json({

